video/out/gl_wayland.c

```diff
@@ -144,6 +152,20 @@ static void egl_create_window(struct vo_wayland_state *wl)
                    wl->egl_context.egl.ctx);
 
     wl_display_dispatch_pending(wl->display.display);
+
+    /**
+     * <http://lists.freedesktop.org/archives/wayland-devel/2013-November/012019.html>
+     *
+     * The main change is that if the swap interval is 0 then Mesa won't install a
+     * frame callback so that eglSwapBuffers can be executed as often as necessary.
+     * Instead it will do a sync request after the swap buffers. It will block for
+     * sync complete event in get_back_bo instead of the frame callback. The
+     * compositor is likely to send a release event while processing the new buffer
+     * attach and this makes sure we will receive that before deciding whether to
+     * allocate a new buffer.
+     */
+
+    eglSwapInterval(wl->egl_context.egl.dpy, 0);
```

Comment:
The GL draw calls still block. And those are done on non wayland related code. I thought this is the most cleanest solution and to make sure no gl calls block.